retirement/tasks.py

```diff
@@ -0,0 +1,36 @@
+from celery import shared_task
+from django.db import transaction
+from django.db.models import Q
+from django.utils import timezone
+from django.contrib.contenttypes.models import ContentType
+
+
+@shared_task
+def assign_retreat_tomatoes():
+    """
+    Assign tomatoes to all active users of a retreat if one of the date passed.
+    """
+    from retirement.models import RetreatDate, Reservation
+    reservation_type = ContentType.objects.get_for_model(
+        Reservation)
+    today = timezone.now()
+    to_assign_dates = RetreatDate.objects.filter(
+        end_time__lte=today,
+        tomatoes_assigned=False,
+    )
+
+    for date in to_assign_dates:
+        with transaction.atomic():
+            active_reservations = date.retreat.reservations.filter(
+                Q(is_active=True) | Q(cancelation_date__gte=date.end_time))
+            for reservation in active_reservations:
+                from tomato.models import Tomato
+                Tomato.objects.get_or_create(
+                    user=reservation.user,
+                    number_of_tomato=date.number_of_tomatoes,
+                    source=Tomato.TOMATO_SOURCE_RETREAT,
+                    content_type=reservation_type,
+                    object_id=reservation.id,
+                )
```

Comment:
Warning here, it's possible that you create multiple row for a tuple user/retreat. You should add a uniqueness on model Meta to make sure it's not possible.